DOC: Tell people to use only-binary option (#21715)
numpy/numpy: doc/source/user/depending_on_numpy.rst

For large, actively maintained packages that depend on NumPy, we recommend
testing against the development version of NumPy in CI. To make this easy,
nightly builds are provided as wheels at
https://anaconda.org/scipy-wheels-nightly/. Example install command::

    pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy

This helps detect regressions in NumPy that need fixing before the next NumPy

> release.  Furthermore, we recommend to raise errors on warnings in CI for this

Reviewer: This recently came up elsewhere, does it make sense to also recommend a way to turn warnings into errors?
```suggestion
release.  Furthermore, we recommend to use `python -Werrors` to turn
warnings into errors in CI for this
```

Author: We use the pytest "filterwarnings" settings for more control (https://docs.pytest.org/en/stable/how-to/capture-warnings.html). `-Werrors` might also turn unrelated warnings into errors, so YMMV. But if you want that anyway, I can apply your suggestion, so please let me know. Thanks!
